test/test_foreach.py

```diff
@@ -740,164 +513,133 @@ def test_add_scalar_with_empty_list_and_empty_tensor(self, device, dtype):
+        foreach_op, foreach_op_ = op.method_variant, op.inplace_variant
+        tensors1 = []
+        tensors2 = []
+
+        # Empty lists
+        with self.assertRaisesRegex(RuntimeError, "There were no tensor arguments to this function"):
+            foreach_op(tensors1, tensors2)
+        with self.assertRaisesRegex(RuntimeError, "There were no tensor arguments to this function"):
+            foreach_op_(tensors1, tensors2)
 
-    @dtypes(*torch.testing.get_all_dtypes())
-    def test_mul_list(self, device, dtype):
-        self._test_bin_op_list(device, dtype, torch._foreach_mul, torch._foreach_mul_, torch.mul)
+        # One empty list
+        tensors1.append(torch.tensor([1], device=device, dtype=dtype))
+        with self.assertRaisesRegex(RuntimeError, "Tensor list must have same number of elements as scalar list."):
+            foreach_op(tensors1, tensors2)
+        with self.assertRaisesRegex(RuntimeError, "Tensor list must have same number of elements as scalar list."):
+            foreach_op_(tensors1, tensors2)
 
-    @dtypes(*torch.testing.get_all_dtypes())
-    def test_div_list(self, device, dtype):
-        if dtype in torch.testing.integral_types_and(torch.bool):
-            if self.device_type == 'cpu':
-                with self.assertRaisesRegex(RuntimeError, "result type Float can't be cast to the desired output type"):
-                    self._test_bin_op_list(device, dtype, torch._foreach_div, torch._foreach_div_, torch.div)
-            else:
-                self.skipTest("Skipped! See https://github.com/pytorch/pytorch/issues/44489")
-            return
+        # Lists have different amount of tensors
+        tensors2.append(torch.tensor([1], device=device))
+        tensors2.append(torch.tensor([1], device=device))
+        with self.assertRaisesRegex(RuntimeError, "Tensor lists must have the same number of tensors, got 1 and 2"):
+            foreach_op(tensors1, tensors2)
+        with self.assertRaisesRegex(RuntimeError, "Tensor lists must have the same number of tensors, got 1 and 2"):
+            foreach_op_(tensors1, tensors2)
 
-        for N in N_values:
-            tensors1 = self._get_test_data(device, dtype, N)
+        # Corresponding tensors with different sizes
+        tensors1 = [torch.zeros(10, 10, device=device, dtype=dtype) for _ in range(10)]
+        tensors2 = [torch.ones(11, 11, device=device, dtype=dtype) for _ in range(10)]
+        with self.assertRaisesRegex(RuntimeError, "Corresponding tensors in lists must have the same size"):
+            foreach_op(tensors1, tensors2)
+        with self.assertRaisesRegex(RuntimeError, r", got \[10, 10\] and \[11, 11\]"):
+            foreach_op_(tensors1, tensors2)
 
-            if dtype in [torch.bfloat16, torch.bool, torch.float16]:
-                tensors2 = [torch.zeros(N, N, device=device, dtype=dtype).add(2) for _ in range(N)]
+        # different devices
+        if torch.cuda.is_available() and torch.cuda.device_count() > 1:
```

Comment:
Please check that `device` argument you got is `cuda`, `torch.device(device).type == "cuda"`